Automattic/pocket-casts-android
# Referrals: add invalid offer screen

## Description
This adds Referrals invalid offer screen

## Testing Instructions
1. Change ReferralsClaimGuestPassBannerCard's `ReferralsPageType.Claim` to `ReferralsPageType.InvalidOffer`
2. Launch the app 
3. Go to Profile tab
4. Tap clam guess pass banner
5. ✅ Notice that invalid offer screen is shown (FVJPaWlRMlosEpsRhqxnSD-fi-2189_8613)

## Screenshots or Screencast 

|Phone - Portrait|
|----|
|<img width=320 src="https://github.com/user-attachments/assets/3a2952ac-2a73-4f21-a25f-b3140fc2847a"/>|

|Phone - Landscape|
|----|
|<img width=400 src="https://github.com/user-attachments/assets/a6a2f46e-a8a0-4f50-b626-7e12fb54e1e1"/>|

|Tablet|
|----|
|<img width=500 src="https://github.com/user-attachments/assets/02e5c904-3217-443e-83f7-57d2a2054f0a"/>|

## Checklist
 
#### I have tested any UI changes...
